op-node/rollup/engine/engine_controller.go

```diff
@@ -75,19 +76,25 @@ type EngineController struct {
 	safeAttrs    *derive.AttributesWithParent
 }
 
-func NewEngineController(engine ExecEngine, log log.Logger, metrics derive.Metrics, rollupCfg *rollup.Config, syncMode sync.Mode) *EngineController {
+func NewEngineController(engine ExecEngine, log log.Logger, metrics derive.Metrics, rollupCfg *rollup.Config, syncCfg *sync.Config) *EngineController {
 	syncStatus := syncStatusCL
-	if syncMode == sync.ELSync {
+	if syncCfg.SyncMode == sync.ELSync {
 		syncStatus = syncStatusWillStartEL
 	}
 
+	engineKind := syncCfg.L2EngineClientKind
+	if !sync.ValidEngineClientKind(syncCfg.L2EngineClientKind) {
+		engineKind = sync.EngineClientGeth
+	}
```

Comment:
(related to the comment I left on `config.go` regarding where this flag should live)

It reads a little strangely for `NewEngineController` to only learn about the engine-kind through the `syncCfg`